pkg/scheduler/factory/factory.go

```diff
@@ -305,11 +306,11 @@ func NewConfigFactory(args *ConfigFactoryArgs) scheduler.Configurator {
 	}
 
 	// Setup cache comparer
-	comparer := &cacheComparer{
-		podLister:  args.PodInformer.Lister(),
-		nodeLister: args.NodeInformer.Lister(),
-		cache:      c.schedulerCache,
-		podQueue:   c.podQueue,
+	comparer := &cachecomparer.CacheComparer{
```

Comment:
Can we add a `NewCacheComparer` func for this, and add a `Run` func to cover the following codes, e.g. signal, go routine?

Reply:
Add a `NewCacheComparer` is always a good idea. But this PR is just move code. I am more inclined to optimize this part of the code in a new PR. What do you think of it? @k82cn